### DISCUSSION TITLE
20230810 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#53] [Meetings]
Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - 
- JCelano(@josecelano) - x 
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by cgbosse** [comment]
# Nautilus 

## Hiring / New Employees

- Arranged the first onboarding meeting for our new colleagues
- Coordinated with the new colleagues to fill out the formal documents
- Got in touch with the labour risk company and organized the training
- Prepared the documentation for the "Welcome pack" folders and the onboarding document
- Helped the mother company with some pending documentation 

## Linkedin

- Networking online and offline

## Torrust

- 

## WP Corp Website:

- 

## Other

- Had some meetings with Cameron Garnham and Jose Celano to discuss various topics
- Prepared the trip to Liechtenstein and Switzerland
- Travelled to Liechtenstein
- Requested the new domain name access to start generating new email accounts for the Nautilus Company with the nautilus-cyberneering.dev domain
- Cameron would like us to create a static page in the nautilus-cyberneering website subdomain using the template from Matt Fantinel

**Comment by josecelano** [comment]
### Torrust

#### Done

**Tracker**

- Error in documentation running tracker with docker: https://github.com/torrust/torrust-tracker/issues/351
- Improve docker UX: https://github.com/torrust/torrust-tracker/issues/361

**Backend**

- Fixed problem with wrong infohash: https://github.com/torrust/torrust-index-backend/issues/242
Run E2E tests using MySQL: https://github.com/torrust/torrust-index-backend/issues/223

**Website**

- [New article: The Enigmatic Torrent "Source" Field](https://torrust.com/the-enigmatic-torrent-source-field)

#### Discussions

##### Which tasks should we finish before the next major release?

I've been working on issues we want to implement before releasing major releases for all the services (tracker, index backend, index frontend). We still have to decide which ones we want to include. I've reviewed all issues in all repos and this is my list:

Common:

1. Refactor: extract a package for common BitTorrent structs such as info-hash, torrent file, etcetera. See https://github.com/torrust/torrust-tracker/issues/360

Index:

- [Allow renaming categories and tags](https://github.com/torrust/torrust-index/issues/104)
- [Allow banning users](https://github.com/torrust/torrust-index/issues/91)

Index Backend:

- [HTTPs support](https://github.com/torrust/torrust-index-backend/issues/131)
- Remove dependency on `intermondal` command, at least to create test torrents (I'm not sure about this). https://github.com/torrust/torrust-index-backend/issues/241

Index Frontend:

- [Add E2E tests](https://github.com/torrust/torrust-index-frontend/issues/148) (5 days)
- [Torrust-Tracker won't compile on Windows](https://github.com/torrust/torrust-tracker/issues/325)
- [Add middleware to compress responses](https://github.com/torrust/torrust-tracker/issues/326)

Website (HTTP://torrust.com)

- Add an introductory blog post: “How to setup the Torrust Tracker-Index for development”. See https://github.com/torrust/torrust-website/issues/58

##### Support for Windows

@mario-nt expressed his concern about this topic, not only because the tracker does not compile for Windows but also because we have a lot of bash scripts. I've added a lot of bash scripts because it was the fastest way for me to add the docker support to setup the E2E testing environments. But in the long term we agreed that we should migrate them to a language for scripting (python/rust) that can run on any OS (and other advantages). This topic is related to the app installation.

For the tracker:

I've collected some links related to the installation:

- https://github.com/torrust/torrust-tracker/discussions/134
- https://github.com/torrust/torrust-tracker/issues/351#issuecomment-1668177192
- https://github.com/torrust/torrust-tracker/discussions/134 

However, most bash scripts are not used to install the application but to setup the E2E environments. For example, if you use MySQL you need to provide a running MySQL service with a given database and the app installation process should only run migrations.

I think we have to differentiate between the system dependencies (what you need to run the app: file system, database, ...) and the application installation (creating the database, seeding database, adding admin user, customizing configuration values, etcetera).

@da2ce7 maybe we can use the [preexisting discussion](https://github.com/torrust/torrust-tracker/discussions/134) to discuss proposals for installation. I think we should first agree on the phases/levels and then on a concrete solution for each part. For example, we could have a rust console command for running the docker container for E2E testing and use a cargo command or web interface for the installation assistant. We will probably need two ways to install the application: one which is more human-friendly and another (for testing) where you can pass the whole configuration you want in a config file, like we are doing right now, to set up the e2e testing evns.

> NOTICE: I put in `./docker/bin/**/*.sh` scripts to setup the system dependencies for E2E testing and in `./bin/*.sh` scripts to setup the E2E testing env at the app level when the infrastructure is already there.

**Reply by josecelano** [reply]
Hey @grmbyrn @mario-nt, since one of you asked why we also supported MySQL, I've remembered another reason to implement the MySQL driver for the torrust projects was the scalability. For example, if you have more than one virtual machine or docker container for the tracker server if you use SQLite, you have to share the file between the machines/containers which is not always easy or allowed by cloud providers. Besides, for large amounts of data, SQLite could not be a valid option.

**Reply by mario-nt** [reply]
Hi @josecelano , @grmbyrn.

About the bash scripts running on windows, I think it is not a problem if the user has installed GIT previously, as if I'm not wrong, it installs the gitbash interpreter. 

Also, I think Windows nowadays is able to run bash commands due to WSL (Windows subsystem for Linux).

I will do some research on this topic and run a few tests, and let you know the result. I think we can include this discussion in the "tracker not running on windows" discussions/issues tab.

**Reply by josecelano** [reply]
Ok, I would like at least (if possible) to compile the app on Windows without the subsystem for Linux. Scripts are still very simple and the installation can be done manually.

**Reply by mario-nt** [reply]
I agree with that, I think WSL is enabled by default (need to check this), but it's better to limit the dependencies as it makes running the app a simpler process.

**Comment by mario-nt** [comment]
### Onboarding/Paperwork

- Welcome/Onboarding meeting
- Setting up communication tools (email, Telegram, Github, Toggle...)
- Reading through mandatory documents 


### Computer and Dev Env Preparation

- Fresh OS install
- Software tools installation (IDEs, git, etc)

### Secure Git Guide

- Reading articles from https://secure-git.guide/
- Research on GPG keys
- Research on signed commits
- Research on encrypted emails for added security
- Research on Github issues/discussions
- Opening of two issues (small bugs and typos)

### Torrust

- First look at website, repositories and documentation
- Reading through blog posts
- Cloning the repositories
- Discussing an issue with the team, about how we can improve the scripts for installing and running the tracker on Windows
- Discussing what features should be included in the next major release

**Comment by grmbyrn** [comment]
## Onboarding
---
* Onboarding meeting
* Reading and signing contracts
* Setting up necessary tools, such as Toggl and Telegram

## Familiarising with Nautilus Repos
---
* Reading documentation for repos, specifically secure git guide, git queue and nautilus librarian
* Visiting and reading through pages for these repos
* Reading through code for any errors or improvements which I could open an issue for.

## Secure Git Guide
---
* Cloning the repo onto my computer so I could view it locally
* Identifying limitations/improvements which I would make
* Opening two issues on these for consideration
* Identify Svelte as a useful framework to migrate this repo to, especially as planning to use with Torrust

## Python
---
* Identify Python as a useful language to learn for future AI projects which could make use of this technology
* Search helpful resources to learn Python which will help me start using it as soon as possible. Freecodecamp's course very comprehensive so far: https://www.youtube.com/watch?v=eWRfhZUzrAc&ab_channel=freeCodeCamp.org

## Diary
---
* Compare different platforms for setting up a diary to document what I am doing for Nautilus. Options included Twitter/X, WordPress and Medium. I opted for Medium in the end as I like the set up, it has a clean, easy to use interface and also has an established community of developers writing there from whom I can learn.
